pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -220,38 +228,32 @@ func (l *podNIC) PlugPhase1() error {
 			return err
 		}
 
-		if err := bindMechanism.preparePodNetworkInterface(); err != nil {
-			log.Log.Reason(err).Error("failed to prepare pod networking")
-			return errors.CreateCriticalNetworkError(err)
+		if l.dhcpConfigurator != nil {
+			dhcpConfig := bindMechanism.generateDhcpConfig()
+			if err := l.dhcpConfigurator.ExportConfiguration(*dhcpConfig); err != nil {
+				log.Log.Reason(err).Error("failed to save dhcpConfig configuration")
+				return errors.CreateCriticalNetworkError(err)
+			}
 		}
 
-		if err := bindMechanism.setCachedDhcpConfig(); err != nil {
-			log.Log.Reason(err).Error("failed to save dhcpConfig configuration")
+		// preparePodNetworkInterface must be called *after* the generate
+		// methods since it mutates the pod interface from which those
+		// generator methods get their info from.
```

Comment:
You wrote in the comment "preparePodNetworkInterface must be called *after* the generate methods".
But `bindMechanism.generateDomainIfaceSpec()` is still called after `preparePodNetworkInterface`. So I believe that even with the comment the situation is still (or even more) confusing.
I would move `bindMechanism.generateDomainIfaceSpec()` to be called before the `prepare`. But it you have a good explanation why it is called after, you can added it as a comment.

Reply:
Updated the comment.